File: zip/zip.c
#include <stdio.h>
#include <math.h>

int main()
{
    int x1, x2,v1, v2;

    scanf("%d",&x1);
    scanf("%d",&v1);
    scanf("%d",&x2);
    scanf("%d",&v2);

    if(v1 > v2 && ((x2-x1)%(v1-v2) == 0))
        printf("YES");
    else
        printf("NO");
}

File: zip/zip.py
x1,v1,x2,v2 = map(int,input().split())
if v1==v2:
    print 'NO'
else:
    if (x1-x2+v2-v1)%(v2-v1)==0:
        if (x1-x2+v2-v1)/(v2-v1)>0:
            print 'YES'
        else:
            print 'NO'
    else:
        print 'NO'
